lib/oneview-sdk/resource.rb

```diff
@@ -327,12 +327,11 @@ def ensure_uri
     # @param [Array<OneviewSDK::Resource>] resources The list of resources
     # @return [Array<String>] List of uris
     # @raise IncompleteResource if 'uri' is not set for each resource.
-    def get_and_ensure_uri_for(resources)
-      uris = resources.map do |resource|
+    def get_and_ensure_uri(resources)
```

Comment:
And while this is a new method, I'd rename it to `ensure_and_get_uris`. This conveys the order of the operations and the plurality of the return values